File: backendRubyOnRails/games/app/models/game.rb
class Game < ApplicationRecord
    before_save :generate_slug

    validates :slug, uniqueness: true

    validates :nom, presence: true
    validates :descripcion, presence: true
    validates :precio, presence: true
    validates :developer, presence: true

    has_many :game_categories, dependent: :destroy
    has_many :categories, through: :game_categories
    has_many :game_images, dependent: :destroy
    def generate_slug
        self.slug = nom.parameterize
    end

    def as_json
        super({ only: %i[slug nom descripcion precio developer isActive ] }).merge(
        categories: categories.pluck(:slug), images: game_images.pluck(:image))
    end

    scope :filter_by_precio, ->(minPrecio, maxPrecio) {
        return all if maxPrecio.nil?
        where("precio >= ? AND precio <= ?", minPrecio || 0, maxPrecio)
    }

    scope :filter_by_search, ->(search) {
        return all if search.nil? || search == ""
        where("LOWER(NOM) LIKE ?", "%#{search.to_s.downcase}%")
    }

    scope :filter_by_categories, ->(categories) {
        return all if categories.nil? || categories == ""
        joins(:categories).where(categories: { slug: categories.split(',') }).distinct
    }

    scope :sort_by_field, ->(sortField,sortAsc = true) {
        return all if sortField.nil? 
        sortOrder= sortAsc == "true" ? :asc : :desc
        case sortField 
        when 'precio'
            order(precio: sortOrder )
        when 'nom'
            order(nom: sortOrder )
        else
            all
        end
    }

end

File: backendGo/games/models/game.go
package models

import (
	"gorm.io/gorm"
)

type Game struct {
	gorm.Model
	Slug           string         `gorm:"uniqueIndex" json:"slug"`
	Nom            string         `json:"nom"`
	Descripcion    string         `json:"descripcion"`
	Precio         int            `json:"precio"`
	Developer      string         `json:"developer"`
	IsActive       bool           `gorm:"default:true;column:isActive" json:"isActive"`
	GameCategories []GameCategory `gorm:"foreignKey:GameID"`
	GameImages     []GameImage    `gorm:"foreignKey:GameID"`
	Categories     []Category     `gorm:"many2many:game_categories;"`
}

func (g *Game) ToMap() map[string]interface{} {

	categories := make([]string, len(g.Categories))
	for i, cat := range g.Categories {
		categories[i] = cat.Nom
	}
	
	images := make([]string, len(g.GameImages))
	for i, img := range g.GameImages {
		images[i] = img.Image
	}

	return map[string]interface{}{
		"slug":        g.Slug,
		"nom":         g.Nom,
		"descripcion": g.Descripcion,
		"precio":      g.Precio,
		"developer":   g.Developer,
		// "isActive":    g.IsActive,
		"categories": categories,
		"images":     images,
	}
}
